addon/-private/system/relationships/relationship-payloads.js

```diff
@@ -44,10 +44,32 @@ export default class RelationshipPayloads {
   constructor(store, modelName, relationshipName, relationshipMeta, inverseModelName, inverseRelationshipName, inverseRelationshipMeta) {
```

Comment:
maybe s/`modelName`/`baseModelName` or something here as we want this to always be the base model name